src/test/java/org/broadinstitute/consent/http/db/DatasetDAOTest.java

```diff
@@ -936,6 +985,166 @@ void testGetApprovedDatasetsWhenEmpty() {
+
+    // ensure we still have no approved datasets.
+    List<ApprovedDataset> approvedDatasets3 = datasetDAO.getApprovedDatasets(user.getUserId());
+    assertEquals(0, approvedDatasets3.size());
+
+    // Simulate 2 DAC 2 elections for today
+    // vote yes on dataset 3
+    Integer electionId5 = electionDAO.insertElection(
+        ElectionType.DATA_ACCESS.getValue(),
+        ElectionStatus.OPEN.getValue(),
+        today,
+        progressReport.getReferenceId(),
+        dataset3.getDatasetId()
+    );
+    Integer voteId5 = voteDAO.insertVote(chairperson2.getUserId(), electionId5, VoteType.FINAL.getValue());
+    voteDAO.updateVote(true, "rationale", today, voteId5, false, electionId5, today, false);
+    electionDAO.updateElectionById(electionId5, ElectionStatus.CLOSED.getValue(), today);
+    assertEquals(1, datasetDAO.getApprovedDatasets(user.getUserId()).size());
+
+    Integer electionId6 = electionDAO.insertElection(
+        ElectionType.DATA_ACCESS.getValue(),
+        ElectionStatus.OPEN.getValue(),
+        today,
+        progressReport.getReferenceId(),
+        dataset4.getDatasetId()
+    );
+    Integer voteId6 = voteDAO.insertVote(chairperson2.getUserId(), electionId6, VoteType.FINAL.getValue());
+    // vote no on dataset 4
+    voteDAO.updateVote(false, "rationale", today, voteId6, false, electionId6, today, false);
+    electionDAO.updateElectionById(electionId6, ElectionStatus.CLOSED.getValue(), today);
+
+    List<ApprovedDataset> approvedDatasets4 = datasetDAO.getApprovedDatasets(user.getUserId());
+    assertNotNull(approvedDatasets4);
+    assertEquals(1, approvedDatasets4.size());
+
+    List<DataAccessRequest> dataset3Dars = dataAccessRequestDAO.findApprovedDARsByDatasetId(dataset3.getDatasetId());
+    assertEquals(1, dataset3Dars.size());
+
+    List<DataAccessRequest> dataset4Dars = dataAccessRequestDAO.findApprovedDARsByDatasetId(dataset4.getDatasetId());
+    assertEquals(0, dataset4Dars.size());
+
+    today = new Date();  //adjust the date into the future or the LAST_VALUE won't work correctly.
+    // make a new election for dataset 3 and now vote no
+    Integer electionId7 = electionDAO.insertElection(
+        ElectionType.DATA_ACCESS.getValue(),
+        ElectionStatus.OPEN.getValue(),
+        today,
+        progressReport.getReferenceId(),
+        dataset3.getDatasetId()
+    );
+    Integer voteId7 = voteDAO.insertVote(chairperson2.getUserId(), electionId7, VoteType.FINAL.getValue());
+    voteDAO.updateVote(false, "rationale", today, voteId7, false, electionId7, today, false);
+    electionDAO.updateElectionById(electionId7, ElectionStatus.CLOSED.getValue(), today);
+    assertEquals(0, datasetDAO.getApprovedDatasets(user.getUserId()).size());
+
+    dataset3Dars = dataAccessRequestDAO.findApprovedDARsByDatasetId(dataset3.getDatasetId());
```

Comment:
This ensures a DAC can hold an election and turn off access to a dataset.  That these now agree is a great thing!